File: src/Python/Selection/maximum.py
class Maximum:
    @staticmethod
    def find_maximum(array):
        """ Returns the maximum value in an array of values. """
        if len(array) == 0:
            return None
        max = array[0] 
        if len(array) == 1:
            return max
        for value in array[1:]:
            if value > max:
                max = value
        return max

File: src/C#/Selection/Maximum.cs
using System;
using System.Collections.Generic;

namespace AlgorithmsLibrary.Selection;

public static class Maximum
{
    /// <summary>
    /// Returns the maximum value in an unsorted collection of <see cref="IComparable"/> values.
    /// </summary>
    public static bool FindMaximum<T>(IEnumerable<T> unsortedValues, out T? maximum) where T : IComparable<T>
    {
        maximum = unsortedValues.FirstOrDefault();

        if (!unsortedValues.Any())
            return false;

        foreach (var value in unsortedValues)
            maximum = Max(maximum, value);

        return true;
    }

    private static T? Max<T>(T? x, T? y) where T : IComparable<T>
    {
        return x?.CompareTo(y) >= 0 ? x : y;
    }
}